src/modules/hashtag/dtos/hashtag-param.dto.ts

```diff
@@ -0,0 +1,18 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { ModelName } from '@src/constants/enum';
+import { IsRecord } from '@src/decorators/is-record.decorator';
+import { IdRequestParamDto } from '@src/dtos/id-request-param.dto';
+import { Type } from 'class-transformer';
+import { Min } from 'class-validator';
+
+export class LessonHashtagParamDto extends IdRequestParamDto {
+  @ApiProperty({
+    description: 'hashtag 고유 ID',
+    type: 'number',
+    required: true,
+  })
+  @IsRecord({ model: ModelName.LessonHashtag, field: 'id' }, true)
+  @Min(1)
```

Comment:
Min 데코레이터에 number 타입 검사가 걸려있어서 다른 데코레이터를 추가하지 않은거같은데 float 형태가 오면 어떻게 되나요??

Reply:
float 형으로 들어오니 prisma에서 Argument id: Got invalid value 1.1 on prisma.findFirstLesson. Provided Float, expected IntFilter or Int에러가 발생하네요 
```suggestion
  @IsRecord({ model: ModelName.LessonHashtag, field: 'id' }, true)
  @Min(1)
  @IsInt()
  @Type(() => Number)
```
class-transformer에서 number변환 후 정수형이 맞는지 확인하는 IsInt를 추가해 주었고,  
IdRequestParamDto의 id에 대한 데코레이터에도 추가해 주겠습니다!

Reply:
현재 IdRequestParamDto에 데코레이터 순서가 
```
  @Min(1)
  @Type(() => Number)
  @IsRecord({}, true)
  id: number;
```
순으로 되어있는데 이렇게 되면 실행 순서가 Type > IsRecord > Min 순으로 실행되어 저희가 의도한데로 IsRecord에 들어갈 id에 대해서 방어가 불가능하게 되어
```
  @IsRecord({}, true)
  @Min(1)
  @IsInt()
  @Type(() => Number)
  id: number;
```
위 처럼 변경하여  Type > IsInt > Min > IsRecord 으로 실행되도록 변경하겠습니다

Reply:
데코레이터 순서에 관계없이 validator 보다 transform 이 먼저 실행돼서 Type 데코레이터 순서는 크게 의미 없지먼 라이브러리에 관계없이 데코레이터 실행 순서를 알기 위해 컨벤션으로 아랭 두는것도 좋겠네요